components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/spark/transformations/DiscardedRowsFilter.java

```diff
@@ -41,11 +41,13 @@ public Boolean call(String[] tokens) throws Exception {
         try {
             Boolean keep = true;
             for (Integer index : indices) {
-                if (index >= tokens.length || MLConstants.EMPTY.equals(tokens[index]) || MLConstants.NA.equals(tokens[index])) {
+                if (index >= tokens.length || MLConstants.EMPTY.equals(tokens[index]) ||
```

Comment:
Can we introduce an enum and use http://stackoverflow.com/a/21730297 to validate?